lib/query/querybuilder.js

```diff
@@ -527,6 +532,12 @@ class Builder extends EventEmitter {
   // Adds a raw `where` clause to the query.
   whereRaw(sql, bindings) {
     const raw = sql.isRawInstance ? sql : this.client.raw(sql, bindings);
+
+    assert(
+      !raw.bindings.some(isObject),
```

Comment:
raw.bindings can be null

Reply:
thanks, updated the assert and tests

Reply:
btw I decided to move the code to MySQL-specific logic since there was a test suite for pg that actually allows arrays as values in bindings https://github.com/antoniopuero/knex/blame/master/test/unit/query/builder.js#L9974 and PG should handle the error correctly. Not sure about other DB providers tho.